lib/descriptors/FieldDescriptor.js

```diff
@@ -131,6 +137,22 @@ FieldDescriptor.prototype.isRepeated = function () {
 }
 
 
+FieldDescriptor.prototype.setOneofIndex = function (oneofIndex) {
+  this._oneofIndex = oneofIndex
```

Comment:
(fwiw, i found this choice at odds with how tags are handled too, though it is handy to have a oneofIndex so we're not forced to pass around oneof name strings everywhere)